core/modules/util/testEventThread.cc

```diff
@@ -0,0 +1,188 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/**
+ *
+ * @brief test EventThread
+ *
+ * @author John Gates, SLAC
+ */
+
+// System headers
+#include <iostream>
+#include <sstream>
+#include <string>
+
+// Third-party headers
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "util/EventThread.h"
+
+// Boost unit test header
+#define BOOST_TEST_MODULE common
+#include "boost/test/included/unit_test.hpp"
+
+namespace test = boost::test_tools;
+
+using namespace lsst::qserv::util;
+
+BOOST_AUTO_TEST_SUITE(Suite)
+
+/** @test
+ * Print a MultiError object containing only one error
+ */
+BOOST_AUTO_TEST_CASE(EventThreadTest) {
+    LOG_DEBUG("EventThread test");
+
+    struct SumUnprotected {
+        int total{0};
+        void add(int val) { total += val; }
+    };
+
+    /// Queue a sum depending on mutex protection within the EventThread.
+    {
+        EventThread et{};
+        SumUnprotected sum;
+        int total{0};
+        for (int j=1; j<10; j++) {
```

Comment:
I have a general problem in this file. Many for loops use hard-coded constants and in many cases they are repeated for the same task. Some the constants seem arbitrary with no explanation. I would suggest defining a const int with the value and explaining what it controls. So, in the future if I want to change it it will change the code everywhere where it needs to change.